src/chunk_append/exec.c

```diff
@@ -366,6 +396,30 @@ choose_next_subplan(ChunkAppendState *state)
 	}
 }
 
+static void
+choose_next_subplan_for_leader(ChunkAppendState *state)
+{
+	state->current = NO_MATCHING_SUBPLANS;
+}
+
+static void
+choose_next_subplan_for_worker(ChunkAppendState *state)
+{
+	LWLockAcquire(state->lock, LW_EXCLUSIVE);
+
+	state->current = state->pstate->last_plan;
```

Comment:
I think the code is a bit confusing here. maybe choose_next_subplan should take the current index as a parameter?